liboptv/src/segmentation.c

```diff
@@ -0,0 +1,535 @@
+/****************************************************************************
+
+Routine:		       	peakfitting.c
+
+Author/Copyright:		Hans-Gerd Maas
+
+Address:		       	Institute of Geodesy and Photogrammetry
+		      		ETH - Hoenggerberg
+		       		CH - 8093 Zurich
+
+Creation Date:			Feb. 1990
+
+Description:
+
+****************************************************************************/
+
+#include "segmentation.h"
+
+/*  targ_rec( ) thresholding and center of gravity with a peak fitting technique  
+      uses 4 neighbours for connectivity and 8 to find local maxima     
+Arguments:
+    unsigned char	*img;		       	 image
+    int         threshold                grey scale threshold 
+    int	       	xmin,xmax,ymin,ymax;	 search area, size of the blog
+    control_par *cpar                   control parameters
+
+Output:
+    target	       	pix[];		       	 pixel coord array
+    int             n_targets           number of targets found
+
+*/
+
+int targ_rec (unsigned char *img, target_par *targ_par, int xmin, 
+int xmax, int ymin, int ymax, control_par *cpar, int num_cam, target pix[]){
+  register int	i, j, m;
+  int	      	n=0, n_wait=0, n_targets=0, sumg, sumg_min;
+  int	        numpix;
+  int	        thres, disco;
+  int	       	cr_sz;			       /* size of crosses to be drawn */
+  int	       	xa,ya,xb,yb, x4[4],y4[4], xn,yn, nx, ny; 
+  double	       	x, y;
+  unsigned char *img0;
+  
+  register unsigned char	gv, gvref;
+    
+  targpix	       	waitlist[2048];
+
+  /* avoid many dereferences */
+  int imx, imy;
+  imx = cpar->imx;
+  imy = cpar->imy;
+
+  thres = targ_par->gvthres[num_cam];
+  disco = targ_par->discont;
+   
+   
+  img0 = (unsigned char *) calloc (imx*imy, 1);
+  /* copy image to a temporary mask */
+  memcpy(img0, img, imx*imy);
+
+  /*  thresholding and connectivity analysis in image  */
+  for (i=ymin; i<ymax; i++)  for (j=xmin; j<xmax; j++)
```

Comment:
I already see a lot of indentation and style fixes needed. I'll do them if I must, but I'll be happy if the next version will already include them :)